File: 345.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        idx = []
        for i, c in enumerate(s):
            if c in 'aeiouAEIOU':
                idx.append(i)
        n = len(idx)
        for i in range(n // 2):
            s[idx[i]], s[idx[n - 1 - i]] = s[idx[n - 1 - i]], s[idx[i]]
        return ''.join(s)

File: sol/345.cpp
struct Solution {
    inline bool isVowel(char c) {
        if (c >= 'a'){
            return (1065233 >> (c - 'a')) & 1;
        } else if (c >= 'A') {
            return (1065233 >> (c - 'A')) & 1;
        } else {
            return false;
        }
    }
    inline string reverseVowels(string &s) {
        int i = 0, j = s.size() - 1;
        while (true){
            while (i < j && !this->isVowel(s[i])){
                ++i;
            }
            while (i < j && !this->isVowel(s[j])){
                --j;
            }
            if (i >= j){
                break;
            }
            swap(s[i++], s[j--]);
        }
        return s;
    }
};
